### PR TITLE
Second Amendment Rights

### PR DESCRIPTION
**Pull request by BobJoga**
## About The Pull Request

A very expansive list of additions and changes to weaponry.
Credits to necromanceranne for the holorifle(https://github.com/tgstation/tgstation/pull/49944)
Credits to hippiestation for a lot, in fact most, that i added.

## Why It's Good For The Game

More makeshift weaponry, more traitor weaponry, means more fun in general. Since this is an MRP-HRP server, we punish powergamers and people who abuse mechanics, as such i see these new additions as reasonable - in fact, a good honeytrap to catch bad players.

## Changelog
:cl: BobJoga, Muhreen, Hippiestation and Necromanceranne
add: The bartender isn't satisfied with how slow knocking people down with his shotgun is. He has made a blueprint for people to upgrade their double barrels to fire both barrels at once.
add: Miners are jeealous of the bartender and his upgraded shotgun. They discovered that with an uranium coating, the same blueprint can be used to upgrade the super combat shotgun they got from bubblegum.
add: Assistants are minorly skilled in robotics, and they now came up with a new mech prototype for the masses: The locker mech. Essentially an useless pile of crap that only serves for entertainment value, and can only take 100 damage.
add: Assistants have now discovered that they can now make shitty laser weaponry with power cells, a bit of know-how, and a bit of dumb luck. These can accept any type of cell which max capacity doesn't exceed 2500, and are quite terrible in accuracy. They can be upgraded with plasteel to diminish this, but not totally get rid of it. They also have a small fire delay per shot.
add: The greytide and some gangsters have felt lacking in luxury. They have come up with a powered deluxe switchblade blueprint to fix that issue. Deals 15 brute damage, sets targets on fire for a small amount.
add: Security feels like their energy weaponry and shotguns aren't metal enough, and have built a working holorifle prototype that is a mix of both. The holorifle deals 40 damage per microfusion cell, + clone damage caused over time.
add: Assistants have discovered that wrapping paper can be used for things other than wrapping presents. With this knowledge, they have developed shanks, makeshift glass hatchets, armored hazard vests and more. Glass hatchets are essentially as good as normal hatchets, armored hazard vests are a teeny bit below durathread vest and deal minor slowdown.
add: The syndicate have been field testing a new weapon with stolen bluespace technology, the contender. The contender is a gun that can fire all kinds of ballistic rounds ever, and functions like a double barreled shotgun. The kit comes packed with 4 .50 bullets (2 normal, 1 penetrator, 1 soporific) as well as a swat helmet and combat knife. Costs 18tc.
add: Miners, enraged with watchers, have developed a new weapon from their corpses capable of shooting freezing beams and charging up with plasma. Essentially, shoots do the same as a watcher freezing beam.
add: The assistants, in their infinite wisdom, have created a weapon that can use all types of casings - at the probable cost of blowing your arm off. It's the box gun. Essentially a contender that can only hold one round, and has a 33% chance to blow your arm off when fired.
add: The assistants are forming a new faction of station protectors and helpers, the minutemen. They've developed a signature weapon, the laser musket. The laser musket works like a pump action energy gun, but can only shoot normal disabler and laser shots. The cell capacity is comparable to the warden's defender.
balance: The syndicate have stopped being cheapskates with their surplus rifle, after too many reported casualties. It is now a viable weapon in combat. It only has a 1 second delay for firing a shot now, and each mag now holds 7 bullets instead of 10.
balance: The syndicate really want to get rid of the cursed ebony blade, since it has been causing way too many cases of friendly fire. The cost of it has been reduced to 14tc.
balance: Botanists are getting better at the art of making durathread clothing. All durathread armor is now more resilient.
fix: The ebony blade and goldbrand now have their proper inhand icons.
imageadd: Icons for the holorifle.
imageadd: Deluxe switchblade icons
imageadd: Hippiestation icons
imageadd: Icons for the hardsuit armblade
/:cl:

### CODE REVIEW

**Comment by BobJoga** [comment]
Currently it only really has a craftable holorifle. More weapons will be added soon.

**Comment by ShadeAware** [comment]
Do I see a **craftable laser rifle with custom debuffs?** :eyes:

**Comment by necromanceranne** [comment]
Hi, I made the original holorifle sprite + coded it. I had an oopsie with the repo so I had to close my pr on /tg/ but if it is alright with you, please at least give credit before using my stuff that I haven't even properly reattempted to use myself. Thank you.

**Comment by BobJoga** [comment]
> Hi, I made the original holorifle sprite + coded it. I had an oopsie with the repo so I had to close my pr on /tg/ but if it is alright with you, please at least give credit before using my stuff that I haven't even properly reattempted to use myself. Thank you.
> 
> (Note if it got merged or denied at that point I don't care but at the very least it would be nice to point back to the original PR I made over here [tgstation/tgstation#49944](https://github.com/tgstation/tgstation/pull/49944))

Yeah, i'm not claiming that this is mine at all really, in fact this is mostly just "stolen" stuff - i'll put credits in for you though!

**Comment by BobJoga** [comment]
this is literally me if this doesn't get merged
doug4

**Comment by HomocidlePancakes** [comment]
Don’t mind me, I am some rando that’s strolling through random SS13 repos. Just wanted to tell you that if you add something in hopes of baiting players in to being banned, then you should reconsider how you play this game. I’ll be taking my leave now.

**Comment by ShadeAware** [comment]
Banbait isn't the focus of the PR. It's just a side effect that Coronatide shutters are gonna try to powergame up the ass and then eat shit.

**Comment by BobJoga** [comment]
> Don’t mind me, I am some rando that’s strolling through random SS13 repos. Just wanted to tell you that if you add something in hopes of baiting players in to being banned, then you should reconsider how you play this game. I’ll be taking my leave now.

Not the focus of the PR, just a "fun" side effect i mentioned. The main focus is here is fun features for normal players which shouldn't be abused.

**Comment by BobJoga** [comment]
literally a week and this PR hasn't been merged nor reviewed in any capacity
haha yes

File: modular_skyrat/code/datums/components/crafting/recipes/recipes_weapon_and_ammo.dm [hunk 1]
```diff
@@ -21,9 +50,74 @@
...
+
+/datum/crafting_recipe/hazardarmor
+	name = "hazard vest armor"
+	result =  /obj/item/clothing/suit/armor/hazard
+	reqs = list(/obj/item/stack/sheet/metal = 1,
+				/obj/item/clothing/suit/hazardvest = 1,
+				/obj/item/stack/wrapping_paper = 5)
+	time = 100
+	category = CAT_CLOTHING
+
```
Comment (KathrinBailey) [inline modular_skyrat/code/datums/components/crafting/recipes/recipes_weapon_and_ammo.dm]: Way too cheap for such a powerful item. **This gives non-security 35 melee armour when stacked with sec/dura jumpsuit.**
**This is your armour block:**
	armor = list(melee = 25, bullet = 10, laser = 0, energy = 0, bomb = 5, bio = 0, rad = 0)
**This is the durathread winter coat armour block:**
	armor = list("melee" = 20, "bullet" = 8, "laser" = 15, "energy" = 8, "bomb" = 25, "bio" = 10, "rad" = 15, "fire" = 75, "acid" = 37)
**This is the durathread vest armour block**
	armor = list("melee" = 20, "bullet" = 10, "laser" = 30, "energy" = 5, "bomb" = 15, "bio" = 0, "rad" = 0, "fire" = 0, "acid" = 50)
**This is the standard security armour (coat/vest)**
	armor = list("melee" = 30, "bullet" = 30, "laser" = 30, "energy" = 10, "bomb" = 25, "bio" = 0, "rad" = 0, "fire" = 50, "acid" = 50)

At the moment, this completely removes the durathread meta. Durathread requires cotton to be mutated and a lot of time to mass-produce, it also require leather and/or cloth to make durathread items. This takes virtually nothing to craft, all the items are in extreme abundance on the station with no access restrictions to get them. Once stacked with durathread, this gives you a hefty 35 melee armour.

Rework options:
• 10 melee armour, 5 bullet.
• 15 melee armour, 5 bullet. More expensive crafting (cable coil+plasteel?)
• 20 melee armour, 10 bullet, slowdown. ('cuz heavy hurrdurr)
• 5 melee armour, 10 to 15 bullet armour. Makes it less useful against NT, more useful againt syndies/nukies. A bit OP still.

Comment (BobJoga) [reply modular_skyrat/code/datums/components/crafting/recipes/recipes_weapon_and_ammo.dm]: I'm gonna take the route of more expensive crafting.

Comment (BobJoga) [reply modular_skyrat/code/datums/components/crafting/recipes/recipes_weapon_and_ammo.dm]: Done. Made durathread more consistent as well.

File: modular_skyrat/code/game/objects/items/misc.dm [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	force = 5
+	throwforce = 5
+	throw_speed = 2
+	throw_range = 3
+	w_class = WEIGHT_CLASS_BULKY
+	attack_verb = list("shoved", "bashed")
+	block_chance = 40
+	var/damage_received = 0 //Amount of damage the shield has received
+	var/max_damage = 100 //Amount of max damage the trayshield can withstand
+
```
Comment (KathrinBailey) [inline modular_skyrat/code/game/objects/items/misc.dm]: Doesn't the wooden buckler have 25 block chance?
The materials required to craft are fairly common. The wooden buckler is more expensive, so this should either be brought in line with it.

Comment (BobJoga) [reply modular_skyrat/code/game/objects/items/misc.dm]: You're kind of right, but it has 30 block chance. I'll make this one have 25 block chance.

Comment (BobJoga) [reply modular_skyrat/code/game/objects/items/misc.dm]: Actually it'll be 30, since it has the disadvantage of breaking very easily

**Comment by KathrinBailey** [comment]
1. **Firing both shotgun barrels at once.** This essentially makes the shotgun into the dual improvised shotgun alpha strike. I personally think it's a bit powerful still but you can accomplish the same thing by other means so +1.
2. **Holorifle** It's pretty cool. I'd consider making this its own weapon and keeping it separate, but it's up to you really. Not sure if it'd see much use if it requires the shotgun, as the shotguns are still at the top of the meta still.
3. **Deluxe Switchblade.** It's more or less a Captain's Sabre but you've swapped delimbing for 3 burn damage and setting targets on fire, but now available to engineering and science, cargo too I think, and anyone who can take stuff out of the ORM.
It's a bit too powerful considering that you can conceal it, might want to drop the brute from 15 to 12 at least.
4. **Locker Mech.** It sounds fun and opens up some future options, such as making improvised weapons into improvised exosuit weapons.
5. **Watcher Gun.** Miners are already the most powerful non-head role in the game that can roll traitor. Arguably more powerful than secoffs. They already have access to 80 melee block chance in a hardsuit, free jetpacks, rwall deleters, instant wall deleters, 80 damage every second in a vacuum, close to that with traitor gear. They also can open cargo chests without access. Temperature guns are insanely powerful and are hard-counters to all opponents wearing hardsuits, or any armour that slows them down. Even if the weapon did no damage, this is something extremely powerful (you can bring people to a halt, making them unable to dodge melee or projectiles) being given to an already extremely powerful role.
6. **Box Gun.** It's a fun idea but it seems way too meme-y for MRP/HRP, you already have plenty of good additions here.
7. **Surplus Rifle.** Seems okay.
8. **Laser Muskets.** Crafting requirements should include two or three T1 capacitors and one or two ignitors so it's a little more difficult to get and it's more of a challenge to scavenge. It'll still be good for maps that don't have a public autolathe, since the public tool storage vendor gives you ignitors and I think even basic parts. You can get basic parts from random machines anyway.

**Comment by BobJoga** [comment]
> 1. **Firing both shotgun barrels at once.** This essentially makes the shotgun into the dual improvised shotgun alpha strike. I personally think it's a bit powerful still but you can accomplish the same thing by other means so +1.
> 2. **Holorifle** It's pretty cool. I'd consider making this its own weapon and keeping it separate, but it's up to you really. Not sure if it'd see much use if it requires the shotgun, as the shotguns are still at the top of the meta still.
> 3. **Deluxe Switchblade.** It's more or less a Captain's Sabre but you've swapped delimbing for 3 burn damage and setting targets on fire, but now available to engineering and science, cargo too I think, and anyone who can take stuff out of the ORM.
>    It's a bit too powerful considering that you can conceal it, might want to drop the brute from 15 to 12 at least.
> 4. **Locker Mech.** It sounds fun and opens up some future options, such as making improvised weapons into improvised exosuit weapons.
> 5. **Watcher Gun.** Miners are already the most powerful non-head role in the game that can roll traitor. Arguably more powerful than secoffs. They already have access to 80 melee block chance in a hardsuit, free jetpacks, rwall deleters, instant wall deleters, 80 damage every second in a vacuum, close to that with traitor gear. They also can open cargo chests without access. Temperature guns are insanely powerful and are hard-counters to all opponents wearing hardsuits, or any armour that slows them down. Even if the weapon did no damage, this is something extremely powerful (you can bring people to a halt, making them unable to dodge melee or projectiles) being given to an already extremely powerful role.
> 6. **Box Gun.** It's a fun idea but it seems way too meme-y for MRP/HRP, you already have plenty of good additions here.
> 7. **Surplus Rifle.** Seems okay.
> 8. **Laser Muskets.** Crafting requirements should include two or three T1 capacitors and one or two ignitors so it's a little more difficult to get and it's more of a challenge to scavenge. It'll still be good for maps that don't have a public autolathe, since the public tool storage vendor gives you ignitors and I think even basic parts. You can get basic parts from random machines anyway.




> 1. **Firing both shotgun barrels at once.** This essentially makes the shotgun into the dual improvised shotgun alpha strike. I personally think it's a bit powerful still but you can accomplish the same thing by other means so +1.
> 2. **Holorifle** It's pretty cool. I'd consider making this its own weapon and keeping it separate, but it's up to you really. Not sure if it'd see much use if it requires the shotgun, as the shotguns are still at the top of the meta still.
> 3. **Deluxe Switchblade.** It's more or less a Captain's Sabre but you've swapped delimbing for 3 burn damage and setting targets on fire, but now available to engineering and science, cargo too I think, and anyone who can take stuff out of the ORM.
>    It's a bit too powerful considering that you can conceal it, might want to drop the brute from 15 to 12 at least.
> 4. **Locker Mech.** It sounds fun and opens up some future options, such as making improvised weapons into improvised exosuit weapons.
> 5. **Watcher Gun.** Miners are already the most powerful non-head role in the game that can roll traitor. Arguably more powerful than secoffs. They already have access to 80 melee block chance in a hardsuit, free jetpacks, rwall deleters, instant wall deleters, 80 damage every second in a vacuum, close to that with traitor gear. They also can open cargo chests without access. Temperature guns are insanely powerful and are hard-counters to all opponents wearing hardsuits, or any armour that slows them down. Even if the weapon did no damage, this is something extremely powerful (you can bring people to a halt, making them unable to dodge melee or projectiles) being given to an already extremely powerful role.
> 6. **Box Gun.** It's a fun idea but it seems way too meme-y for MRP/HRP, you already have plenty of good additions here.
> 7. **Surplus Rifle.** Seems okay.
> 8. **Laser Muskets.** Crafting requirements should include two or three T1 capacitors and one or two ignitors so it's a little more difficult to get and it's more of a challenge to scavenge. It'll still be good for maps that don't have a public autolathe, since the public tool storage vendor gives you ignitors and I think even basic parts. You can get basic parts from random machines anyway.

**Replies:**
2. Shotguns are already top of the meta, but i really like the idea of this being a shotgun and energy gun bastard hybrid. So i'll keep it like that.
3. I think it's fair enough as-is, it's just somewhat better than a survival knife. Best i can do is take off the burn damage and just let it ignite people.
5. 80 melee block chance? I think you meant armor. Even still, getting the real gamer shit as miner is either based on hard challenges or RNG. I think the best option here is making the projector require more stuff from RnD, as well as requiring argent cells.
6. I just like how silly the idea is... I know it's terribly memey, but holy hell is it funny. I don't know if i should remove it (yet), i'll wait for the say of a maintainer.
8. Agreed. I'll make that change.

**Comment by BobJoga** [comment]
Actually yeah, i'll make the holorifle a new thing entirely

**Comment by BobJoga** [comment]
alright i'm fucking done, i'm separating this into 20 fucking PRs or whatever

**Comment by BobJoga** [comment]
fucking hell no too is too tiring even for me

**Comment by BobJoga** [comment]
2 more days without merging and then i actually fucking do it i'm gonna spam the shit out of this github and bloat the code like afucking balloon

**Comment by RatFromTheJungle** [comment]
very cool, thank you kanye!

**Comment by BobJoga** [comment]
maintainers PLEASE FOR FUCKS SAKE MERGE THIS ALREADY I SPENT SO MUCH FUCKING TIME ON THIS AND GET NOT EVEN A FUCKING REVIEW FUCK

fuckinghell

**Comment by BobJoga** [comment]
i am going to stick a fucking sharp rod into my rectum

**Comment by BobJoga** [comment]
**PR's closed due to mutes caused by Kathrin Bailey. For any grievances or complaints, contact Floof Ball#0798 on discord. The PR owner will also not listen to any of your complaints unless you provide screenshots that prove you have contacted Floof Ball#0798 on discord first.**
pool

**Pull request closed without merging**